feat: use % thresholds for `MemoryHeapCheck` and `MemoryRSSCheck`

theoludwig opened the pull request:
<!---
Please carefully read the contribution docs before creating a pull request
 👉 https://github.com/adonisjs/.github/blob/main/docs/CONTRIBUTING.md
-->

### 🔗 Linked issue

<!-- Please ensure there is an open issue and mention its number as #123 -->

Fixes #3 

### ❓ Type of change

<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->

- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

<!-- Describe your changes in detail -->
<!-- Why is this change required? What problem does it solve? -->
<!-- If it resolves an open issue, please link to the issue here. For example "Resolves #1337" -->

Add `warnWhenExceedsPercentage` and `failWhenExceedsPercentage` methods to `MemoryRSSCheck` and `MemoryHeapCheck`.

### 📝 Checklist

<!-- Put an `x` in all the boxes that apply. -->
<!-- If your change requires a documentation PR, please link it appropriately -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

Is another PR in https://github.com/adonisjs/v6-docs needed to update the documentation?

theoludwig committed:
feat!: use % thresholds for `MemoryHeapCheck`

BREAKING CHANGE:
Update `warnWhenExceeds` and `failWhenExceeds` functions signatures to
accept a percentage instead of a value in bytes.

theoludwig committed:
feat!: use % thresholds for `MemoryRSSCheck`

BREAKING CHANGE:
Update `warnWhenExceeds` and `failWhenExceeds` functions signatures to
accept a percentage instead of a value in bytes.

thetutlage commented:
It seems like we removed the support for specifying the limit in bytes. I would like to support both. The percentage and the bytes.

theoludwig committed:
feat: use bytes thresholds for `MemoryHeapCheck`

theoludwig committed:
feat: use bytes thresholds for `MemoryRSSCheck`

theoludwig commented:
@thetutlage 

> It seems like we removed the support for specifying the limit in bytes. I would like to support both. The percentage and the bytes.

Yes, that's what I said in the PR description:
> BREAKING CHANGE:
Update `warnWhenExceeds` and `failWhenExceeds` functions signatures to accept a percentage instead of a value in bytes.
Both for `MemoryHeapCheck` and `MemoryRSSCheck`.

> As you're already working on a new major version v3, I thought why not doing it, in a non-backwards compatible way (I hope that it will not hurt too much consumers of this package), the only change needed is to update `warnWhenExceeds('300 mb')` with a percentage instead.

---

However, if you want to keep backwards compatibility for that, sure, I updated the PR, so now there are 2 new methods:
- `warnWhenExceedsPercentage`
- `failWhenExceedsPercentage`

thetutlage commented:
Thanks 👍🏽

theoludwig commented:
@thetutlage Could you publish a new version including this change, please? e.g: v3.0.0-next.1

thetutlage commented:
Done. https://github.com/adonisjs/health/releases/tag/v3.1.0-next.0
